internal/ingress/metric/collector/collector_test.go

```diff
@@ -30,6 +30,7 @@ func TestNewUDPLogListener(t *testing.T) {
 	fn := func(message []byte) {
 		t.Logf("message: %v", string(message))
 		atomic.AddUint64(&count, 1)
+		time.Sleep(time.Millisecond)
```

Comment:
Why is this sleep necessary? I would suggest to rather work with channels here if you are looking to test that this function is called exactly once. This seems like it will be flaky when done based on sleeping.